### PR TITLE
the commit where i make some relics

### PR DESCRIPTION
**Pull request by nightbug**
adds the relics:
medic pouch, pocketknife, fencing gloves, feather boots, samurai sword

### CODE REVIEW

File: src/main/java/the_warlord/powers/ParryPower.java [hunk 1]
```diff
@@ -77,7 +81,8 @@ public void atStartOfTurn() {
         if (isParrying) {
             ParryDeck.setParried(true);
             AbstractPlayer p = AbstractDungeon.player;
-
+            if(p.hasRelic(FencingGloves.ID)){ addToBot(new ApplyPowerAction(p, p, new ReactionTimePower(p, 1))); }
+            if(p.hasRelic(SamuraiSword.ID)){ addToBot(new GainEnergyAction(1)); }
             //this calls onParry for all powers that implement onParrySubsciber
```
Comment (Cartopol) [inline src/main/java/the_warlord/powers/ParryPower.java:86]: For these types of things I created the onParrySubscriber interface. Every time you parry, I call onParry() on all powers and all cards that have implemented this interface. So instead of adding these functions to the ParryPower, it is done in an onParry method of the power or card itself. 
I haven't implemented this system yet for relics, and I don't mind doing that part. But if you want to learn how this works, check out a power like QuickDrawPower. To do the same for relics, you would have to add the following stuff in the ParryPower:
`for (AbstractRelic relic : p.relics) {
if (relic instanceof OnParrySubscriber) {
relic.onParry(isFullParrying) }
}`
In other words, for each relic, check if it implements the OnParrySubscriber interface. If it does, call the relic's onParry method. 

This kind of subscriber model makes it much easier to implement functionality in the class that should have it, instead of implementing it in methods of other classes.

**Review by Cartopol — CHANGES_REQUESTED** [review]
Looks really good, Squeeny! There's an improvement that can be made to use the OnParrySubscriber interface for relics, if you're up for it. Other than that, looks really good!

**Comment by nightbug** [comment]
pushed a commit with the relevant changes

**Review by Cartopol — APPROVED** [review]
